contrib/completion/git-prompt.sh

```diff
@@ -111,33 +111,46 @@
 __git_printf_supports_v=
```

Comment:
> ```
> I will not look at this series futher during the current development
> cycle that is about to conclude
> ```

Sure, I did not try to time this patchset. Whenever someone has the bandwidth to handle it, it's fine by me. We're not in a hurry.

(and I'd assume the same about the reply to my comments below)

> ```
> Just forbid use of $'\octal' notation in the coding
> guidelines document, and implement just one variant.
> ```

Yeah. That's fair.

But I'd prefer to examine existing cases of $'...' at the git code base, if there are such, before deciding on a policy. I did not look at existing git scripts.

The two subjects I'd care about are performance and maintinability. I don't know if we want every script do this juggling of one printf and then extracting values from it, and even a single additional command substitution in a hot script can be non-negligible.

Depending on how the scripting ecosystem looks like, it might be most managable (maybe also most performant) to have a single file which sets C0 control chars values __git_001, __git_002 ... __git_037 either literally or however else we decide, then any script which needs control literals can dot it, or embed it, etc. The point is to not have every script reinventing the wheel.

I will update this commit to remove the fallback system (not right away, I prefer to wait for hoepfully more comments)

Reply:
@avih please send this reply as an email instead. GitGitGadget [is _not_ mirroring emails <-> PR comments bidirectionally](https://github.com/gitgitgadget/gitgitgadget/issues/154). It only mirrors emails -> PR comments, **not** vice versa.

Reply:
> @avih please send this reply as an email instead

Yes, I understood after the 1st time you wrote it above. But I already replied to both messages when I saw it.

Thanks again.